### ISSUE TITLE
[Accessibility] [Omnichannel]  problems trying to navigate across messages to be narrated by the screen reader

### ISSUE THREAD

**Issue by elopezanaya** [microsoft/BotFramework-WebChat#4100]
Environment Details:
Application Name: Microsoft Dynamics 365 Omnichannel Online Chat – CRA Online Chat Agent
#URL: Omnichannel Agent Dashboard
Windows Version: Win10
Chromium edge Version 98.0.1089.1 (Official build) dev (64-bit)

## Describe the bug

When the user tabs to the minimized chat window and opens it after hearing that a message has been received, the screen reader does not automatically start reading the messages they received.

Additional to the problem is not possible to navigate across messages using tab or arrows, making impossible for the narrator to read the messages

## Steps to reproduce

1. Start Narrator/jaws.
2. Hit the URL "Omnichannel Agent Dashboard" and login with appropriate credentials
3. Tab to Omnichannel Administration app and hit enter
4. Tab to "chat" menu item and hit enter, select a chat and tab to code snippet under general settings.
5. Open url "aka.ms/omniwidget" and paste the code from code snippet .
6. Tab to the Let's chat button and initiate a conversation
7. Hit URL: Omnichannel Agent Dashboard and log in with appropriate credentials in another browser as an agent.
8. Tab till accept button in the new chat dialog and press enter.
9. From the agent chat, reply the message from the customer
10. From the customer side , open the chat
11. put the focus on the box with the messages
12. try to navigate using arrows or tabs across messages

## Expected behavior

Narrator should start to read the new messages send by the agent.
Navigation with tab and arrows should be supported, once the user put the focus in one mssage it should be easy to navigate up or down 

## Additional context

[Bug]

**Comment by tdurnford** [comment]
@compulim Any updates on this issue?

**Comment by carlosscastro** [comment]
Hey @compulim could you please pick up this issue? Thanks!

**Comment by msomanathan** [comment]
@compulim Could you please provide an update to this issue?

**Comment by compulim** [comment]
Can you verify the version of Web Chat used?

> To determine what version of Web Chat you are running, open your browser's development tools, and paste the following line of code into the console.
>
> `[].map.call(document.head.querySelectorAll('meta[name^="botframework-"]'), function (meta) { return meta.outerHTML; }).join('\n')`

Transcript navigation is a feature we introduced in 4.12.0 (2021 March).

The feature enables end-user to press <kbd>SHIFT</kbd> + <kbd>TAB</kbd> to the transcript and use up/down arrow keys to navigate the transcript.

**Comment by elopezanaya** [comment]
for arrow navigation -> 

I did another try in akams/ocdailym , using 3 narrators (NVDA, JAWS, NARRATOR)->

Windows narrator , NVDa and JAWS, the only one where the functionality was supported was supported was for NVDA

CustomProperties: "{\"requestId\":\"oovi39\",\"request\":{\"url\":\"https://unqed5ece4e719e4765a0602c30b44eb-crm.oc.crmlivetie.com/presence/SendHeartbeat/\",\"requestType\":\"PUT\",\"header\":{\"Cache-Control\":\"no-cache\",\"Ocp-Apim-Trace\":true,\"Content-Type\":\"application/json\",\"OrganizationId\":\"ed5ece4e-719e-4765-a060-2c30b44ebadd\",\"TransactionId\":\"28eec89a-154a-4151-a9ba-157f8db713cf\"},\"staticData\":{\"requestType\":\"CCProxy\",\"requestId\":\"oovi39\"},\"payload\":\"{\\\"AgentId\\\":\\\"19ae8bfa-bf66-ec11-8f8f-000d3a320e31\\\",\\\"ClientSessionId\\\":\\\"qi622h\\\"}\"}}"

**Comment by compulim** [comment]
After clarifying with Edgar, the expected behavior is: "after a new message come, the user should be able to jump to the first unread message fairly quickly,"

It could be a shortcut key to jump to the first unread message. It could also be, when the user is focusing on the "End of chat history" prompt, they can tap on the prompt to jump to the first unread message.

We will discuss about which accessibility experience best fit our customers.

**Comment by compulim** [comment]
After discussing with Michelle from accessibility team today, the expected behavior should left unchanged.

When the Omnichannel chat widget is restored from minimized, all new messages should be read. Thus, all messages on the screen will be marked as read.

Having a shortcut key, or some sort of mechanisms, to jump to the first unread activity, will always land on the last activity in the chat history as everything is read. This is the reason why the expected behavior should kept unchanged.

As the original expected behavior is no repro on latest version of Web Chat, I am closing the issue.
